src/Groundwater.jl

```diff
@@ -39,35 +40,37 @@ function run_timestep(c::Aquifer, tt::Int)
   v = c.Variables
   p = c.Parameters
   d = c.Dimensions
-  # computation of lateral flows:
-  v.lateralflows[:,tt]=zeros(d.aquifers[end],1)
-  for aa in 1:d.aquifers[end]
-    for aa_ in (aa+1):(d.aquifers[end]-1)
-      if p.aquiferconnexion[aa,aa_]==1.
-        if tt==1
-        latflow = p.lateralconductivity[aa,aa_]*(p.piezohead0[aa_]-p.piezohead0[aa])*p.deltatime; # in m3/month or m3/year if factor 12
-        else
-        latflow = p.lateralconductivity[aa,aa_]*(v.piezohead[aa_,tt-1]-v.piezohead[aa,tt-1])*p.deltatime; # in m3/month or m3/year if factor 12
-        end
-        v.lateralflows[aa,tt] += latflow/1000;
-        v.lateralflows[aa_,tt] += -latflow/1000;
-      end
-    end
+  ## initialization
+  if tt==1
+	  head = p.piezohead0;
+  else
+	  head = v.piezohead[:,tt-1]; 
   end
+  flow = zeros(d.aquifers[end],1);
+  v.lateralflows[:,tt] = flow;
+  ## repeat simulation timestep time
+  for mm in 1:config["timestep"]
+  
+  	# computation of lateral flows:
+  	lflows=zeros(d.aquifers[end],1)
+  	for aa in 1:d.aquifers[end]
+    		for aa_ in (aa+1):(d.aquifers[end]-1)
+      			if p.aquiferconnexion[aa,aa_]==1.
+				latflow = p.lateralconductivity[aa,aa_]*(head[aa_]-head[aa]); # in m3/month
+				lflows[aa] += latflow/1000;
+				v.lateralflows[aa,tt] += latflow/1000;
+				lflows[aa_] += -latflow/1000;
+				v.lateralflows[aa_,tt] -= latflow/1000;
```

Comment:
lflows are the monthly lateral flows, which we use to compute head change each month and which are reset to 0 at each iteration.
The variable "lateralflows" is the total lateral flows that are taking place during one simulation time step, which are the sum of all lflows. I have changed it to make it clearer.